guix-science-nonfree/packages/nvidia-container-toolkit.scm

```diff
@@ -0,0 +1,203 @@
+	    (lambda* (#:key inputs outputs #:allow-other-keys)
+	      (invoke "ar" "rcs" "_out/Linux_x86_64/libnvidia-modprobe-utils.a" "_out/Linux_x86_64/nvidia-modprobe-utils.o" "_out/Linux_x86_64/pci-sysfs.o")
+	      (copy-recursively "_out/Linux_x86_64/" (string-append (assoc-ref %outputs "out") "/lib"))))
+	  (delete 'check)
+	  (add-after 'patch-source-shebangs 'replace-prefix
+	    (lambda* (#:key inputs outputs #:allow-other-keys)
+    	      (setenv "CC" "gcc")
+	      (setenv "PREFIX" (assoc-ref %outputs "out"))
+	      (copy-recursively "modprobe-utils/" (string-append (assoc-ref %outputs "out") "/include"))
+	      #true) ; must return true for success
+	    ))
+      #:tests? #f))
+    (native-inputs
+     (list gcc-toolchain m4))
+    (synopsis "Load the NVIDIA kernel module and create NVIDIA character device files")
+    (description "Load the NVIDIA kernel module and create NVIDIA character device files")
+    (home-page "https://github.com/NVIDIA/nvidia-modprobe")
+    (license gpl2)))
+
+(define-public libnvidia-container
+  (package
+    (name "libnvidia-container")
+    (version "1.13.1")
+    (source (origin
+              (method git-fetch)
+              (uri (git-reference
+                    (url "https://github.com/NVIDIA/libnvidia-container")
+                    (commit (string-append "v" version))))
+              (file-name (git-file-name name version))
+      	      (patches (search-patches "libnvidia-container.patch"))
+              (sha256
+               (base32
+		"0rzvh1zhh8pi5xjzaq3nmyzpcvjy41gq8w36dp1ai11a6j2lpa99"))))
+    (build-system gnu-build-system)
+    (arguments
+     (list
+      #:phases
+      #~(modify-phases %standard-phases
+	  (delete 'configure)
+	  (delete 'build)
+	  (delete 'check)
+          (add-after 'unpack 'ensure-writable-source
+	    (lambda* (#:key inputs outputs #:allow-other-keys)
+      	      (setenv "HOME" "/tmp")
+	      (make-file-writable "src/ldcache.c")
+      	      (make-file-writable "src/ldcache.h")
+              (make-file-writable "src/nvc_info.c")))
+	  (add-after 'patch-source-shebangs 'replace-prefix
+	    (lambda* (#:key inputs outputs #:allow-other-keys)
+	      (substitute* "Makefile"
+		(("/usr/local") (assoc-ref outputs "out"))  ;this overrides the prefix
+		(("debug??libdir?") "debug")  ;ensure debug files get installed in the correct subdir
+		((".*nvidia-modprobe.mk.*") "\n")
+		(("^all: shared static tools")
+		 "all: shared tools")
+		((".*LIB_STATIC.*libdir.*$") ""))
+     	      (substitute* "mk/nvcgo.mk"
+		((".*-rf.*")
+		 "\tmkdir -p ${SRCS_DIR} && echo \"sources dir: ${SRCS_DIR}\"\n")
+		(("CURDIR./src/..PREFIX.")
+		 "CURDIR)/src/$(PREFIX)/*")) ;deleting sources fails
+              (substitute* "src/cli/libnvc.c"
+		(("libnvidia-ml.so.1")
+		 "/run/current-system/profile/lib/libnvidia-ml.so.1"))
+              (substitute* "src/nvc_internal.h"
+		(("libnvidia-ml.so.1")
+		 "/run/current-system/profile/lib/libnvidia-ml.so.1"))
```

Comment:
This will only work on Guix System, with nvidia installed globally; I wonder if this can be avoided?

Reply:
is there a standard way of detecting whether we're on guix system or a foreign distro? I could make this subistitutes conditional on that. This might also affect the patch since that also refers to /run/current-system/profile/lib.